lib/ecto/adapters/mysql/connection.ex

```diff
@@ -316,7 +316,12 @@ if Code.ensure_loaded?(Mariaex.Connection) do
       raise ArgumentError, "Array type is not supported by MySQL"
     end
 
-    defp expr(%Ecto.Query.Tagged{value: other, type: type}, sources) when type in [:integer, :float] do
+    defp expr(%Ecto.Query.Tagged{value: binary, type: :binary}, _sources) when is_binary(binary) do
```

Comment:
@mobileoverlord this was here in the past. Then we removed and we added back. If you haven't removed it, you should be fine.